apis/speech/v1alpha1/phraseset_types.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var SpeechPhraseSetGVK = GroupVersion.WithKind("SpeechPhraseSet")
+
+// SpeechPhraseSetSpec defines the desired state of SpeechPhraseSet
+// +kcc:proto=google.cloud.speech.v2.PhraseSet
+type SpeechPhraseSetSpec struct {
+	// The SpeechPhraseSet name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	Parent `json:",inline"`
+
+	// A list of word and phrases.
+	// +kcc:proto:field=google.cloud.speech.v2.PhraseSet.phrases
+	Phrases []PhraseSet_Phrase `json:"phrases,omitempty"`
+
+	// Hint Boost. Positive value will increase the probability that a specific
+	//  phrase will be recognized over other similar sounding phrases. The higher
+	//  the boost, the higher the chance of false positive recognition as well.
+	//  Valid `boost` values are between 0 (exclusive) and 20. We recommend using a
+	//  binary search approach to finding the optimal value for your use case as
+	//  well as adding phrases both with and without boost to your requests.
+	// +kcc:proto:field=google.cloud.speech.v2.PhraseSet.boost
+	Boost *float32 `json:"boost,omitempty"`
```

Comment:
Float is discouraged by controller-gen, could we use `*string`?

Reply:
Done
https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/4303/commits/f94687ff11be8ff0fd4795c1ba630f5798383704